code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -336,76 +336,139 @@
-			NewFlora = list(/obj/structure/flora/grass/green/style_random, /obj/structure/flora/grass/brown/style_random, /obj/structure/flora/grass/both/style_random)
-		if("jungle") //Beneficial due to actually having breathable air. Plus, monkeys and bows and arrows.
-			NewTerrainFloors = /turf/open/floor/grass
-			NewTerrainWalls = /turf/closed/wall/mineral/wood
-			NewTerrainChairs = /obj/structure/chair/wood
-			NewTerrainTables = /obj/structure/table/wood
-			NewFlora = list(/obj/structure/flora/bush/sparsegrass/style_random, /obj/structure/flora/bush/ferny/style_random, /obj/structure/flora/bush/leavy/style_random,
-							/obj/structure/flora/bush/grassy/style_random, /obj/structure/flora/bush/sunny/style_random, /obj/structure/flora/tree/palm/style_random, /mob/living/carbon/human/species/monkey)
-			florachance = 20
-		if("ayy lmao") //Beneficial, turns stuff into alien alloy which is useful to cargo and research. Also repairs atmos.
-			NewTerrainFloors = /turf/open/floor/plating/abductor
-			NewTerrainWalls = /turf/closed/wall/mineral/abductor
-			NewTerrainChairs = /obj/structure/bed/abductor //ayys apparently don't have chairs. An entire species of people who only recline.
-			NewTerrainTables = /obj/structure/table/abductor
-
 /obj/machinery/anomalous_crystal/theme_warp/ActivationReaction(mob/user, method)
-	if(..())
-		var/area/A = get_area(src)
-		if(!A.outdoors && !(A in affected_targets))
-			for(var/atom/Stuff in A)
-				if(isturf(Stuff))
-					var/turf/T = Stuff
-					if((isspaceturf(T) || isfloorturf(T)) && NewTerrainFloors)
-						var/turf/open/O = T.ChangeTurf(NewTerrainFloors, flags = CHANGETURF_IGNORE_AIR)
-						if(prob(florachance) && NewFlora.len && !O.is_blocked_turf(TRUE))
-							var/atom/Picked = pick(NewFlora)
-							new Picked(O)
-						continue
-					if(iswallturf(T) && NewTerrainWalls)
-						T.ChangeTurf(NewTerrainWalls)
-						continue
-				if(istype(Stuff, /obj/structure/chair) && NewTerrainChairs)
-					var/obj/structure/chair/Original = Stuff
-					var/obj/structure/chair/C = new NewTerrainChairs(Original.loc)
-					C.setDir(Original.dir)
-					qdel(Stuff)
-					continue
-				if(istype(Stuff, /obj/structure/table) && NewTerrainTables)
-					new NewTerrainTables(Stuff.loc)
-					continue
-			affected_targets += A
+	. = ..()
+	if (!.)
+		return
+	var/area/current_area = get_area(src)
+	if (current_area in affected_targets)
+		return
+	if (terrain_theme.transform_area(current_area))
+		affected_targets += current_area
+
+/// Transforms an area's turfs and objects into a different theme
+/datum/crystal_warp_theme
+	/// Friendly name of theme
+	var/name
+	/// Typepath of floor to replace open turfs with
+	var/floor
+	/// Typepath fo wall to replace closed turfs with
+	var/wall
+	/// Typepath of object to replace chairs with
+	var/chair
+	/// Typepath of object to replace tables with
+	var/table
+	/// Typepath of wildlife to potentially spawn on transformed open turfs
+	var/list/wildlife = list()
```

Comment:
Not a fan of the wildlife term because that implies mobs, which isn't what you have later on... Maybe `var/list/flora_and_fauna`? Unsure. Could be fine as well but at first glance I'm feeling 😐